geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/netty/ExecutionHandlerContext.java

```diff
@@ -234,7 +235,10 @@ private RedisResponse getExceptionResponse(ChannelHandlerContext ctx, Throwable
       }
     }
 
-    if (cause instanceof NumberFormatException) {
+
+    if (cause instanceof RedisException) {
+      response = RedisResponse.error(cause.getMessage());
+    } else if (cause instanceof NumberFormatException) {
       response = RedisResponse.error(cause.getMessage());
     } else if (cause instanceof ArithmeticException) {
       response = RedisResponse.error(cause.getMessage());
```

Comment:
These three cases all have the same behaviour (calling `RedisResponse.error(cause.getMessage())`) which is already found further down in this if/else block. Would it make sense to combine all the checks into one case:
```
    } else if (cause instanceof IllegalStateException
        || cause instanceof RedisParametersMismatchException
        || cause instanceof RedisException
        || cause instanceof NumberFormatException
        || cause instanceof ArithmeticException) {
      response = RedisResponse.error(cause.getMessage());
```

Reply:
Good suggestion. I've done that and also noticed a couple of other exceptions that can be removed (related to function execution which we are not doing anymore).